app/models/allocation.rb

```diff
@@ -25,5 +45,16 @@ class Allocation < VersionedModel
   validates :moves_count, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :date, presence: true
 
+  # TODO: re-enable once cancellation reason is implemented in FE
+  # validates :cancellation_reason, inclusion: { in: CANCELLATION_REASONS }, if: :cancelled?
+  # validates :cancellation_reason, absence: true, unless: :cancelled?
+
   attribute :complex_cases, Types::JSONB.new(Allocation::ComplexCaseAnswers)
+
+  def cancel
+    self.status = ALLOCATION_STATUS_CANCELLED
+    self.moves.each { |move| move.assign_attributes(status: Move::MOVE_STATUS_CANCELLED) }
```

Comment:
cancelled moves _must_ have a `cancellation_reason` (otherwise validations will fail), which must be one of these values:
```
    CANCELLATION_REASON_MADE_IN_ERROR = 'made_in_error',
    CANCELLATION_REASON_SUPPLIER_DECLINED_TO_MOVE = 'supplier_declined_to_move',
    CANCELLATION_REASON_REJECTED = 'rejected',
    CANCELLATION_REASON_OTHER = 'other'
```